### PR TITLE
vim: simplify substitution in whole file

### PR DESCRIPTION
**Pull request by mstruebing**
- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines

### CODE REVIEW

File: pages/common/vim.md [hunk 1]
```diff
@@ -23,9 +23,9 @@
 
 `<Esc>/{{search_pattern}}<Enter>`
 
-- Perform a regex substitution in the whole file (from the start, `1`, to the end, `$`):
+- Perform a regex substitution in the whole file (from the start to the end(% holds the file name)):
 
-`<Esc>:1,$s/{{pattern}}/{{replacement}}/g<Enter>`
+`<Esc>:%/{{pattern}}/{{replacement}}/g<Enter>`
```
Comment (agnivade) [inline pages/common/vim.md]: % => %s

File: pages/common/vim.md [hunk 2]
```diff
@@ -23,9 +23,9 @@
 
 `<Esc>/{{search_pattern}}<Enter>`
 
-- Perform a regex substitution in the whole file (from the start, `1`, to the end, `$`):
+- Perform a regex substitution in the whole file (from the start to the end(% holds the file name)):
```
Comment (agnivade) [inline pages/common/vim.md]: We can remove the (from start to the end) part. That was needed to signify 1 -> $.

File: pages/common/vim.md (hunk 1 above)
Comment (mstruebing) [reply pages/common/vim.md]: Sorry you are right :)
I've overssen it. Will fix it now.

File: pages/common/vim.md (hunk 2 above)
Comment (ericbn) [reply pages/common/vim.md]: I like this change, but I would remove everything under parenthesis. In this case it is not true that "% holds the file name". Actually it is a [range](http://vimdoc.sourceforge.net/htmldoc/cmdline.html#:range), and the most commonly used one.

Comment (mstruebing) [reply pages/common/vim.md]: I thought holds the file name. When I type `:!echo %` it's also printing the file name.
But I've tried right now that I can't do smth like `:<file>s/pattern/pattern/modifier` - which makes sense. 
Fixed.

Comment (ericbn) [reply pages/common/vim.md]: @mstruebing, the vim documentation is very good. With [`:help :s`](http://vimdoc.sourceforge.net/htmldoc/change.html#:s) you see that it's actually:

    :[range]s[ubstitute]/{pattern}/{string}/[flags] [count]

Comment (agnivade) [reply pages/common/vim.md]: @mstruebing - Can you push your commit ? I dont see any updates.

**Comment by mstruebing** [comment]
@agnivade Now, I've pushed to the wrong repo after renaming the `tldr-go-client` to `tldr` and my fork of this repo to `tldr-pages` because the remote was still the old one :)

**Comment by agnivade** [comment]
ping @sbrl

**Comment by sbrl** [comment]
`vim` isn't my strong point, but looks ok to me!
Thanks, @mstruebing! :smiley_cat:

**Pull request merged**